### PR TITLE
tidb_query: Migrate `json_merge` from `builtin_json.rs` to tidb_query

### PR DESCRIPTION
**Pull request by mapleFU**
###  What have you changed?

* Migrate implementation for json_array from `builtin_json.rs` to the new `impl_json.rs`. This pr is split from pr #5253 .
* Add support for `JsonMergeSig` in `rpn_expr/mod.rs` for `components/tidb_query`.

Don't assume reviewers understand the original issue.

###  What is the type of the changes?

- New feature (Support new Rpn Function )

###  How is the PR tested?

New unit tests.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

File: components/tidb_query/src/rpn_expr/impl_json.rs [hunk 1]
```diff
@@ -27,6 +27,27 @@ fn json_array(args: &[&Option<Json>]) -> Result<Option<Json>> {
...
+#[rpn_fn(varg, min_args = 2)]
+#[inline]
+pub fn json_merge(args: &[&Option<Json>]) -> Result<Option<Json>> {
+    // min_args = 2, so it's ok to call args[0]
+    let mut base_json = match args[0] {
+        None => return Ok(None),
+        Some(json) => json.to_owned(),
+    };
+
+    for json_to_merge in &args[1..] {
```
Comment (iosmanthus) [inline components/tidb_query/src/rpn_expr/impl_json.rs]: Try to use `fold`?

Comment (mapleFU) [reply components/tidb_query/src/rpn_expr/impl_json.rs]: > Try to use `fold`?

Using it now.

**Pull request merged**